### PR TITLE
superscope: Fix center channel generation for spectrum

### PR DESCRIPTION
**Pull request by hartwork**
# Input Left + Right
avs_left avs_right

# Center Output
##  Before
avs_center_broken

## After
avs_center_fixed

---
Created with https://github.com/hartwork/visdriver and https://github.com/KDE/spectacle :smiley:

### CODE REVIEW

**Comment by grandchild** [comment]
unfortunately this seems to break the waveform in a similar way. 🙁 

left & right look the same, of course, but center clearly wraps:

center

**Comment by hartwork** [comment]
@grandchild it looked fine over here :thinking: Let me have a closer look :+1:

**Comment by hartwork** [comment]
@grandchild fixed, better now?

**Comment by grandchild** [comment]
This fixes it, indeed! Cool 🙂 

I now want this fixed everywhere, because some other effects do the same thing and make the same mistake. I'll have to dig out which effects exactly do this. Also I think this code is now sufficiently complex to start thinking about introducing a common utils function. 🤔

For posterity: This bug is present in the original AVS as well, you just never notice it, because normally the left & right spectrum are pretty close together. But with such artificial FFTs like above it shows. I made [a special audio file](https://github.com/grandchild/vis_avs/files/10784280/noise-brownian-left-purple-right.zip) merged from [this](https://en.wikipedia.org/wiki/File:Brown_noise.ogg) on the left channel and [this](https://en.wikipedia.org/wiki/File:Purple_noise.ogg) on the right channel to test this in regular Winamp:

spec-overflow

(`noise-brownian-left-purple-right.wav` is CC-BY-SA as per the original files.)

**Comment by hartwork** [comment]
@grandchild sounds good! Let's decouple things a bit though to not have this unfixed and blocked until its fixed everywhere at once. What do you think?

**Comment by grandchild** [comment]
```
❯ grep -rl "center_channel"
vis_avs/r_oscring.cpp
vis_avs/r_oscstar.cpp
vis_avs/e_simple.cpp
vis_avs/e_superscope.cpp
```
seems like it's only these 4. and the EEL gunctions `getspec()` doesn't have this problem (https://github.com/grandchild/vis_avs/blob/effect-api-port/avs/vis_avs/avs_eelif.cpp#L83)

**Comment by grandchild** [comment]
I have no preference whether you fix those files or I do, I just would like the changes to be in one commit. If you're fine with it, I could amend your commit with the rest and add myself to `Co-authored-by:` or you take the other 3 as well.

What do you think?

**Comment by grandchild** [comment]
Of course, if I'd amend your commit you'd still get to see it first and get final say on it.

File: avs/vis_avs/e_superscope.cpp
Comment (grandchild) [inline avs/vis_avs/e_superscope.cpp:115]: Please add braces here -- I know they're not present in the original (an oversight on my part), but while we're touching it, might as well add them.

File: avs/vis_avs/e_superscope.cpp
Comment (grandchild) [inline avs/vis_avs/e_superscope.cpp:120]: Same as above, please add braces.

**Review by grandchild — CHANGES_REQUESTED** [review]
Please use the prefix `[bug] SuperScope: ` for your commit message, thanks.

File: avs/vis_avs/e_superscope.cpp
Comment (hartwork) [reply avs/vis_avs/e_superscope.cpp]: @grandchild alright, fixed and pushed

**Pull request merged**
